packages/components/src/components/swipeable-notification/swipeable-notification.tsx

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import debounce from 'lodash.debounce';
+import classNames from 'classnames';
+import { NavLink } from 'react-router-dom';
+import { CSSTransition } from 'react-transition-group';
+import { useSwipeable } from 'react-swipeable';
+import { Localize } from '@deriv/translations';
+import Text from '../text';
+
+type TSwipeableNotificationProps = React.PropsWithChildren<{
+    classname?: string;
+    is_failure?: boolean;
+    is_success?: boolean;
+    onUnmount?: () => void;
+    redirect_to?: string;
+    timestamp?: number | null;
+    visibility_duration_ms?: number;
+}>;
+
+const DIRECTION = {
+    LEFT: 'left',
+    RIGHT: 'right',
+};
+
+const SwipeableNotification = ({
+    children,
+    classname = 'swipeable-notification',
+    is_failure,
+    is_success,
+    onUnmount,
+    redirect_to = '',
+    timestamp,
+    visibility_duration_ms,
+}: TSwipeableNotificationProps) => {
+    const [swipe_direction, setSwipeDirection] = React.useState(DIRECTION.LEFT);
+    const [is_visible, setIsVisible] = React.useState(true);
+    const getSeconds = React.useCallback((timestamp?: number | null) => {
+        return timestamp ? Math.abs(Math.floor(Date.now() / 1000) - timestamp) : null;
+    }, []);
+    const [seconds, setSeconds] = React.useState<number | null>(getSeconds(timestamp));
+    const interval_ref = React.useRef<ReturnType<typeof setInterval>>();
+
+    const hideNotification = () => setIsVisible(false);
+    const debouncedHideNotification = React.useMemo(
+        () => debounce(hideNotification, visibility_duration_ms),
+        [visibility_duration_ms]
+    );
+    const getDisplayedTime = () => {
+        if (!seconds && seconds !== 0) return;
+        return seconds < 10 ? (
+            <Localize i18n_default_text='now' />
+        ) : (
+            <Localize i18n_default_text='{{seconds}}s ago' values={{ seconds }} />
+        );
+    };
+    const onSwipeLeft = () => {
+        setSwipeDirection(DIRECTION.LEFT);
+        hideNotification();
+    };
+    const onSwipeRight = () => {
```

Comment:
May be it is better to do one function instead of `onSwipeLeft` and `onSwipeRight`? It will receive direction arg (`DIRECTION.LEFT` or `DIRECTION.RIGHT`)